Build failure: nsjail

### Steps To Reproduce

Steps to reproduce the behavior:
1. build nsjail

### Build log

```
nsjail>   642 |     void* elements[(std::numeric_limits<int>::max() - 2 * sizeof(int)) /
nsjail>       |           ^~~~~~~~
nsjail> cc1plus: all warnings being treated as errors
nsjail> make: *** [<builtin>: config.o] Error 1
nsjail> make: *** Waiting for unfinished jobs....
error: builder for '/nix/store/ax1vxmplpmahyf05ldcy1mwjzbc9yjk3-nsjail-3.4.drv' failed w
ith exit code 2;
       last 25 log lines:
       >     inlined from 'bool config::parseInternal(nsjconf_t*, const nsjail::NsJailCo
       >     inlined from 'bool config::parseFile(nsjconf_t*, const char*)' at config.cc
       > /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/proto
       >   272 |     return *cast<TypeHandler>(rep_->elements[index]);
       >       |             ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
       > /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/proto
       > /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/proto
       >   642 |     void* elements[(std::numeric_limits<int>::max() - 2 * sizeof(int))
       >       |           ^~~~~~~~
       > In member function 'const typename TypeHandler::Type& google::protobuf::interna
       >     inlined from 'const Element& google::protobuf::RepeatedPtrField<T>::Get(int
       >     inlined from 'const nsjail::IdMap& nsjail::NsJailConfig::_internal_uidmap(i
       >     inlined from 'const nsjail::IdMap& nsjail::NsJailConfig::uidmap(int) const'
       >     inlined from 'bool config::parseInternal(nsjconf_t*, const nsjail::NsJailCo
       >     inlined from 'bool config::parseFile(nsjconf_t*, const char*)' at config.cc
       > /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/proto
       >   272 |     return *cast<TypeHandler>(rep_->elements[index]);
       >       |             ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
       > /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/proto
       > /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/proto
       >   642 |     void* elements[(std::numeric_limits<int>::max() - 2 * sizeof(int))
       >       |           ^~~~~~~~
       > cc1plus: all warnings being treated as errors
       > make: *** [<builtin>: config.o] Error 1
       > make: *** Waiting for unfinished jobs....
       For full logs, run 'nix log /nix/store/ax1vxmplpmahyf05ldcy1mwjzbc9yjk3-nsjail-3.
┏━ 1 Errors:
 ⋮
┃        >     inlined from 'const nsjail::IdMap& nsjail::NsJailConfig::uidmap(int) cons
┃        >     inlined from 'bool config::parseInternal(nsjconf_t*, const nsjail::NsJail
┃        >     inlined from 'bool config::parseFile(nsjconf_t*, const char*)' at config.
┃        > /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/pro
┃        >   272 |     return *cast<TypeHandler>(rep_->elements[index]);
┃        >       |             ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
┃        > /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/pro
┃        > /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/pro
┃        >   642 |     void* elements[(std::numeric_limits<int>::max() - 2 * sizeof(int)
┃        >       |           ^~~~~~~~
┃        > cc1plus: all warnings being treated as errors
┃        > make: *** [<builtin>: config.o] Error 1
┃        > make: *** Waiting for unfinished jobs....
┃        For full logs, run 'nix log /nix/store/ax1vxmplpmahyf05ldcy1mwjzbc9yjk3-nsjail…
┣━ Dependency Graph:
┃ ⚠ nsjail-3.4 failed with exit code 2 after ⏱ 3s in buildPhase
┣━━━
┗━ ∑ ⚠ Exited after 1 build failures at 16:48:17 after 4s
yt in 🌐 bee in nixpkgs on  master [$] took 4s
❯ : hx .
yt in 🌐 bee in nixpkgs on  master [$] took 43s
❯ : nom-build -A nsjail
this derivation will be built:
  /nix/store/jnbgzc0r29bk1dkd9ly1hp1svlprlkr2-nsjail-3.4.drv
nsjail> building '/nix/store/jnbgzc0r29bk1dkd9ly1hp1svlprlkr2-nsjail-3.4.drv'
nsjail> Running phase: unpackPhase
nsjail> unpacking source archive /nix/store/zjvjhymwyh484lghpsbry2yw9mvdcxv8-source
nsjail> source root is source
nsjail> Running phase: patchPhase
nsjail> Running phase: updateAutotoolsGnuConfigScriptsPhase
nsjail> Running phase: configurePhase
nsjail> no configure script, doing nothing
nsjail> Running phase: buildPhase
nsjail> build flags: -j16 SHELL=/nix/store/9vafkkic27k7m4934fpawl6yip3a6k4h-bash-5.2-p21
nsjail> LDFLAGS="" CFLAGS=-fPIE make -C kafel
nsjail> make[1]: Entering directory '/build/source/kafel'
nsjail> Makefile:27: warning: overriding recipe for target 'test'
nsjail> build/Makefile.mk:41: warning: ignoring old recipe for target 'test'
nsjail> make -C src PROJECT_ROOT=../
nsjail> make[2]: Entering directory '/build/source/kafel/src'
nsjail> flex lexer.l
nsjail> bison parser.y
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
nsjail> bison parser.y
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
en -std=gnu11 -I../include -Wall -Wextra -Werror -O2   -c -o syscalls/arm_syscalls.o sys
calls/arm_syscalls.c
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
in/newuidmap -DNEWGIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/
bin/newgidmap -O2 -c -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -fPIE -Wformat -Wformat-securi
ty -Wno-format-nonliteral -Wall -Wextra -Werror -Ikafel/include -DPROTOBUF_USE_DLLS -DNO
nsjail> protoc --cpp_out=. config.proto
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
in/newuidmap -DNEWGIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/
bin/newgidmap -O2 -c -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -fPIE -Wformat -Wformat-securi
ty -Wno-format-nonliteral -Wall -Wextra -Werror -Ikafel/include -DPROTOBUF_USE_DLLS -DNO
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
in/newuidmap -DNEWGIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/
bin/newgidmap -O2 -c -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -fPIE -Wformat -Wformat-securi
ty -Wno-format-nonliteral -Wall -Wextra -Werror -Ikafel/include -DPROTOBUF_USE_DLLS -DNO
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
in/newuidmap -DNEWGIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/
bin/newgidmap -O2 -c -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -fPIE -Wformat -Wformat-securi
ty -Wno-format-nonliteral -Wall -Wextra -Werror -Ikafel/include -DPROTOBUF_USE_DLLS -DNO
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
in/newuidmap -DNEWGIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/
bin/newgidmap -O2 -c -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -fPIE -Wformat -Wformat-securi
ty -Wno-format-nonliteral -Wall -Wextra -Werror -Ikafel/include -DPROTOBUF_USE_DLLS -DNO
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
in/newuidmap -DNEWGIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/
bin/newgidmap -O2 -c -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -fPIE -Wformat -Wformat-securi
ty -Wno-format-nonliteral -Wall -Wextra -Werror -Ikafel/include -DPROTOBUF_USE_DLLS -DNO
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
en -std=gnu11 -I../include -Wall -Wextra -Werror -O2 -Wl,-soname,../libkafel.so.1 -share
d kafel.o context.o codegen.o expression.o includes.o parser_types.o policy.o range_rule
s.o syscall.o lexer.o parser.o syscalls/amd64_syscalls.o syscalls/i386_syscalls.o syscal
nsjail> ld  -r kafel.o context.o codegen.o expression.o includes.o parser_types.o policy
.o range_rules.o syscall.o lexer.o parser.o syscalls/amd64_syscalls.o syscalls/i386_sysc
alls.o syscalls/aarch64_syscalls.o syscalls/mipso32_syscalls.o syscalls/mips64_syscalls.
o syscalls/riscv64_syscalls.o syscalls/arm_syscalls.o -o libkafel_r.o
nsjail> # Fix x86 PIC symbols
nsjail> objdump -t libkafel_r.o | grep '.hidden' | grep -v 'get_pc_thunk' | sed -e 's/^.
*\.hidden //' > libkafel.localize_symbols
nsjail> objcopy -w --localize-symbols=libkafel.localize_symbols libkafel_r.o libkafel.o
nsjail> rm -f libkafel.localize_symbols
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
nsjail> rm -f libkafel_r.o
nsjail> ar rcs ../libkafel.a libkafel.o
nsjail> rm -f libkafel.o
nsjail> make[2]: Leaving directory '/build/source/kafel/src'
nsjail> make[1]: Leaving directory '/build/source/kafel'
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
in/newuidmap -DNEWGIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/
bin/newgidmap -O2 -c -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -fPIE -Wformat -Wformat-securi
ty -Wno-format-nonliteral -Wall -Wextra -Werror -Ikafel/include -DPROTOBUF_USE_DLLS -DNO
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
nsjail> config.cc:305:13: error: variable or field 'logHandler' declared void
nsjail>   305 | static void logHandler(
nsjail>       |             ^~~~~~~~~~
nsjail> config.cc:306:23: error: 'LogLevel' is not a member of 'google::protobuf'; did y
ou mean 'nsjail::LogLevel'?
nsjail>   306 |     google::protobuf::LogLevel level, const char* filename, int line, co
nst std::string& message) {
nsjail>       |                       ^~~~~~~~
nsjail> In file included from config.cc:40:
nsjail> config.pb.h:107:6: note: 'nsjail::LogLevel' declared here
nsjail>   107 | enum LogLevel : int {
nsjail>       |      ^~~~~~~~
nsjail> config.cc:306:39: error: expected primary-expression before 'const'
nsjail>   306 |     google::protobuf::LogLevel level, const char* filename, int line, co
nsjail>       |                                       ^~~~~
nsjail> config.cc:306:61: error: expected primary-expression before 'int'
nsjail>   306 |     google::protobuf::LogLevel level, const char* filename, int line, co
nsjail>       |                                                             ^~~
nsjail> config.cc:306:71: error: expected primary-expression before 'const'
nsjail>   306 |     google::protobuf::LogLevel level, const char* filename, int line, co
nsjail>       |                                                                       ^~
nsjail> config.cc: In function 'bool config::parseFile(nsjconf_t*, const char*)':
nsjail> config.cc:319:23: error: 'logHandler' was not declared in this scope; did you me
nsjail>   319 |         SetLogHandler(logHandler);
nsjail>       |                       ^~~~~~~~~~
nsjail>       |                       sighandler_t
nsjail> config.cc:319:9: error: 'SetLogHandler' was not declared in this scope
nsjail>   319 |         SetLogHandler(logHandler);
nsjail>       |         ^~~~~~~~~~~~~
nsjail> make: *** [<builtin>: config.o] Error 1
nsjail> make: *** Waiting for unfinished jobs....
error: builder for '/nix/store/jnbgzc0r29bk1dkd9ly1hp1svlprlkr2-nsjail-3.4.drv' failed w
ith exit code 2;
       last 25 log lines:
       >   306 |     google::protobuf::LogLevel level, const char* filename, int line, c
       >       |                       ^~~~~~~~
       > In file included from config.cc:40:
       > config.pb.h:107:6: note: 'nsjail::LogLevel' declared here
       >   107 | enum LogLevel : int {
       >       |      ^~~~~~~~
       > config.cc:306:39: error: expected primary-expression before 'const'
       >   306 |     google::protobuf::LogLevel level, const char* filename, int line, c
       >       |                                       ^~~~~
       > config.cc:306:61: error: expected primary-expression before 'int'
       >   306 |     google::protobuf::LogLevel level, const char* filename, int line, c
       >       |                                                             ^~~
       > config.cc:306:71: error: expected primary-expression before 'const'
       >   306 |     google::protobuf::LogLevel level, const char* filename, int line, c
       >       |                                                                       ^
       > config.cc: In function 'bool config::parseFile(nsjconf_t*, const char*)':
       > config.cc:319:23: error: 'logHandler' was not declared in this scope; did you m
       >   319 |         SetLogHandler(logHandler);
       >       |                       ^~~~~~~~~~
       >       |                       sighandler_t
       > config.cc:319:9: error: 'SetLogHandler' was not declared in this scope
       >   319 |         SetLogHandler(logHandler);
       >       |         ^~~~~~~~~~~~~
       > make: *** [<builtin>: config.o] Error 1
       > make: *** Waiting for unfinished jobs....
       For full logs, run 'nix log /nix/store/jnbgzc0r29bk1dkd9ly1hp1svlprlkr2-nsjail-3.
┏━ 1 Errors:
┏━ 1 Errors:
 ⋮
┃        > config.cc:306:71: error: expected primary-expression before 'const'
┃        >   306 |     google::protobuf::LogLevel level, const char* filename, int line…
┃        >       |                                                                     …
┃        > config.cc: In function 'bool config::parseFile(nsjconf_t*, const char*)':
 mean 'sighandler_t'?
┃        >   319 |         SetLogHandler(logHandler);
┃        >       |                       ^~~~~~~~~~
┃        >       |                       sighandler_t
┃        > config.cc:319:9: error: 'SetLogHandler' was not declared in this scope
┃        >   319 |         SetLogHandler(logHandler);
┃        >       |         ^~~~~~~~~~~~~
┃        > make: *** [<builtin>: config.o] Error 1
┃        > make: *** Waiting for unfinished jobs....
┃        For full logs, run 'nix log /nix/store/jnbgzc0r29bk1dkd9ly1hp1svlprlkr2-nsjail…
┣━ Dependency Graph:
┃ ⚠ nsjail-3.4 failed with exit code 2 after ⏱ 3s in buildPhase
┗━ ∑ ⚠ Exited after 1 build failures at 16:49:59 after 4s
yt in 🌐 bee in nixpkgs on  master [$] took 4s
❯ : gitui
yt in 🌐 bee in nixpkgs on  master [$] took 5s
❯ : nom-build -A nsjail
this derivation will be built:
  /nix/store/ax1vxmplpmahyf05ldcy1mwjzbc9yjk3-nsjail-3.4.drv
nsjail> building '/nix/store/ax1vxmplpmahyf05ldcy1mwjzbc9yjk3-nsjail-3.4.drv'
nsjail> Running phase: unpackPhase
nsjail> unpacking source archive /nix/store/zjvjhymwyh484lghpsbry2yw9mvdcxv8-source
nsjail> source root is source
nsjail> Running phase: patchPhase
nsjail> Running phase: updateAutotoolsGnuConfigScriptsPhase
nsjail> Running phase: configurePhase
nsjail> no configure script, doing nothing
nsjail> Running phase: buildPhase
nsjail> build flags: -j16 SHELL=/nix/store/9vafkkic27k7m4934fpawl6yip3a6k4h-bash-5.2-p21
nsjail> LDFLAGS="" CFLAGS=-fPIE make -C kafel
nsjail> make[1]: Entering directory '/build/source/kafel'
nsjail> Makefile:27: warning: overriding recipe for target 'test'
nsjail> build/Makefile.mk:41: warning: ignoring old recipe for target 'test'
nsjail> make -C src PROJECT_ROOT=../
nsjail> make[2]: Entering directory '/build/source/kafel/src'
nsjail> flex lexer.l
nsjail> bison parser.y
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
nsjail> bison parser.y
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
in/newuidmap -DNEWGIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/
bin/newgidmap -O2 -c -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -fPIE -Wformat -Wformat-securi
ty -Wno-format-nonliteral -Wall -Wextra -Werror -Ikafel/include -I/nix/store/f5nrc42s6g7
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
nsjail> protoc --cpp_out=. config.proto
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
en -std=gnu11 -I../include -Wall -Wextra -Werror -O2   -c -o kafel.o kafel.c
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
en -std=gnu11 -I../include -Wall -Wextra -Werror -O2 -Wno-error   -c -o lexer.o lexer.c
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
in/newuidmap -DNEWGIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/
bin/newgidmap -O2 -c -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -fPIE -Wformat -Wformat-securi
ty -Wno-format-nonliteral -Wall -Wextra -Werror -Ikafel/include -I/nix/store/f5nrc42s6g7
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
in/newuidmap -DNEWGIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/
bin/newgidmap -O2 -c -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -fPIE -Wformat -Wformat-securi
ty -Wno-format-nonliteral -Wall -Wextra -Werror -Ikafel/include -I/nix/store/f5nrc42s6g7
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
in/newuidmap -DNEWGIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/
bin/newgidmap -O2 -c -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -fPIE -Wformat -Wformat-securi
ty -Wno-format-nonliteral -Wall -Wextra -Werror -Ikafel/include -I/nix/store/f5nrc42s6g7
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
in/newuidmap -DNEWGIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/
bin/newgidmap -O2 -c -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -fPIE -Wformat -Wformat-securi
ty -Wno-format-nonliteral -Wall -Wextra -Werror -Ikafel/include -I/nix/store/f5nrc42s6g7
nsjail> gcc -fPIE -std=gnu11 -Iinclude -Wall -Wextra -Werror -O2 -fPIC -fvisibility=hidd
en -std=gnu11 -I../include -Wall -Wextra -Werror -O2 -Wl,-soname,../libkafel.so.1 -share
d kafel.o context.o codegen.o expression.o includes.o parser_types.o policy.o range_rule
s.o syscall.o lexer.o parser.o syscalls/amd64_syscalls.o syscalls/i386_syscalls.o syscal
nsjail> ld  -r kafel.o context.o codegen.o expression.o includes.o parser_types.o policy
.o range_rules.o syscall.o lexer.o parser.o syscalls/amd64_syscalls.o syscalls/i386_sysc
alls.o syscalls/aarch64_syscalls.o syscalls/mipso32_syscalls.o syscalls/mips64_syscalls.
o syscalls/riscv64_syscalls.o syscalls/arm_syscalls.o -o libkafel_r.o
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
nsjail> # Fix x86 PIC symbols
nsjail> objdump -t libkafel_r.o | grep '.hidden' | grep -v 'get_pc_thunk' | sed -e 's/^.
nsjail> objcopy -w --localize-symbols=libkafel.localize_symbols libkafel_r.o libkafel.o
nsjail> rm -f libkafel.localize_symbols
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
in/newuidmap -DNEWGIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/
bin/newgidmap -O2 -c -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -fPIE -Wformat -Wformat-securi
nsjail> rm -f libkafel_r.o
nsjail> ar rcs ../libkafel.a libkafel.o
nsjail> rm -f libkafel.o
nsjail> g++ -DNEWUIDMAP_PATH=/nix/store/599hpidqm9g1mfjyank0wnrzx5dycxr7-shadow-4.14.2/b
nsjail> make[2]: Leaving directory '/build/source/kafel/src'
nsjail> make[1]: Leaving directory '/build/source/kafel'
nsjail> In file included from /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12
/include/google/protobuf/repeated_field.h:60,
nsjail>                  from /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12
/include/google/protobuf/implicit_weak_message.h:39,
nsjail>                  from /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12
nsjail>                  from /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12
nsjail>                  from /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12
nsjail>                  from /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12
nsjail>                  from /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12
nsjail>                  from config.cc:26:
nsjail> In member function 'const typename TypeHandler::Type& google::protobuf::internal
nsjail>     inlined from 'const Element& google::protobuf::RepeatedPtrField<T>::Get(int)
nsjail>     inlined from 'const nsjail::MountPt& nsjail::NsJailConfig::_internal_mount(i
nsjail>     inlined from 'const nsjail::MountPt& nsjail::NsJailConfig::mount(int) const'
nsjail>     inlined from 'bool config::parseInternal(nsjconf_t*, const nsjail::NsJailCon
fig&)' at config.cc:226:32,
nsjail>     inlined from 'bool config::parseFile(nsjconf_t*, const char*)' at config.cc:
331:20:
nsjail> /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/protob
nsjail>   272 |     return *cast<TypeHandler>(rep_->elements[index]);
nsjail>       |             ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
nsjail> /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/protob
nsjail> /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/protob
nsjail>   642 |     void* elements[(std::numeric_limits<int>::max() - 2 * sizeof(int)) /
nsjail>       |           ^~~~~~~~
nsjail> In member function 'const typename TypeHandler::Type& google::protobuf::internal
nsjail>     inlined from 'const Element& google::protobuf::RepeatedPtrField<T>::Get(int)
nsjail>     inlined from 'const nsjail::MountPt& nsjail::NsJailConfig::_internal_mount(i
nsjail>     inlined from 'const nsjail::MountPt& nsjail::NsJailConfig::mount(int) const'
nsjail>     inlined from 'bool config::parseInternal(nsjconf_t*, const nsjail::NsJailCon
nsjail>     inlined from 'bool config::parseFile(nsjconf_t*, const char*)' at config.cc:
nsjail> /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/protob
nsjail>   272 |     return *cast<TypeHandler>(rep_->elements[index]);
nsjail>       |             ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
nsjail> /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/protob
nsjail> /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/protob
nsjail>   642 |     void* elements[(std::numeric_limits<int>::max() - 2 * sizeof(int)) /
nsjail>       |           ^~~~~~~~
nsjail> In member function 'const typename TypeHandler::Type& google::protobuf::internal
nsjail>     inlined from 'const Element& google::protobuf::RepeatedPtrField<T>::Get(int)
nsjail>     inlined from 'const nsjail::IdMap& nsjail::NsJailConfig::_internal_gidmap(in
nsjail>     inlined from 'const nsjail::IdMap& nsjail::NsJailConfig::gidmap(int) const'
nsjail>     inlined from 'bool config::parseInternal(nsjconf_t*, const nsjail::NsJailCon
nsjail>     inlined from 'bool config::parseFile(nsjconf_t*, const char*)' at config.cc:
nsjail> /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/protob
nsjail>   272 |     return *cast<TypeHandler>(rep_->elements[index]);
nsjail>       |             ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
nsjail> /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/protob
nsjail> /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/protob
nsjail>   642 |     void* elements[(std::numeric_limits<int>::max() - 2 * sizeof(int)) /
nsjail>       |           ^~~~~~~~
nsjail> In member function 'const typename TypeHandler::Type& google::protobuf::internal
nsjail>     inlined from 'const Element& google::protobuf::RepeatedPtrField<T>::Get(int)
nsjail>     inlined from 'const nsjail::IdMap& nsjail::NsJailConfig::_internal_gidmap(in
nsjail>     inlined from 'const nsjail::IdMap& nsjail::NsJailConfig::gidmap(int) const'
nsjail>     inlined from 'bool config::parseInternal(nsjconf_t*, const nsjail::NsJailCon
nsjail>     inlined from 'bool config::parseFile(nsjconf_t*, const char*)' at config.cc:
nsjail> /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/protob
nsjail>   272 |     return *cast<TypeHandler>(rep_->elements[index]);
nsjail>       |             ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
nsjail> /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/protob
nsjail> /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/protob
nsjail>   642 |     void* elements[(std::numeric_limits<int>::max() - 2 * sizeof(int)) /
nsjail>       |           ^~~~~~~~
nsjail> In member function 'const typename TypeHandler::Type& google::protobuf::internal
nsjail>     inlined from 'const Element& google::protobuf::RepeatedPtrField<T>::Get(int)
nsjail>     inlined from 'const nsjail::IdMap& nsjail::NsJailConfig::_internal_gidmap(in
nsjail>     inlined from 'const nsjail::IdMap& nsjail::NsJailConfig::gidmap(int) const'
nsjail>     inlined from 'bool config::parseInternal(nsjconf_t*, const nsjail::NsJailCon
nsjail>     inlined from 'bool config::parseFile(nsjconf_t*, const char*)' at config.cc:
nsjail> /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/protob
nsjail>   272 |     return *cast<TypeHandler>(rep_->elements[index]);
nsjail>       |             ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
nsjail> /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/protob
nsjail> /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/protob
nsjail>   642 |     void* elements[(std::numeric_limits<int>::max() - 2 * sizeof(int)) /
nsjail>       |           ^~~~~~~~
nsjail> In member function 'const typename TypeHandler::Type& google::protobuf::internal
nsjail>     inlined from 'const Element& google::protobuf::RepeatedPtrField<T>::Get(int)
nsjail>     inlined from 'const nsjail::IdMap& nsjail::NsJailConfig::_internal_uidmap(in
nsjail>     inlined from 'const nsjail::IdMap& nsjail::NsJailConfig::uidmap(int) const'
nsjail>     inlined from 'bool config::parseInternal(nsjconf_t*, const nsjail::NsJailCon
nsjail>     inlined from 'bool config::parseFile(nsjconf_t*, const char*)' at config.cc:
nsjail> /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/protob
nsjail>   272 |     return *cast<TypeHandler>(rep_->elements[index]);
nsjail>       |             ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
nsjail> /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/protob
nsjail> /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/protob
nsjail>   642 |     void* elements[(std::numeric_limits<int>::max() - 2 * sizeof(int)) /
nsjail>       |           ^~~~~~~~
nsjail> In member function 'const typename TypeHandler::Type& google::protobuf::internal
nsjail>     inlined from 'const Element& google::protobuf::RepeatedPtrField<T>::Get(int)
nsjail>     inlined from 'const nsjail::IdMap& nsjail::NsJailConfig::_internal_uidmap(in
nsjail>     inlined from 'const nsjail::IdMap& nsjail::NsJailConfig::uidmap(int) const'
nsjail>     inlined from 'bool config::parseInternal(nsjconf_t*, const nsjail::NsJailCon
nsjail>     inlined from 'bool config::parseFile(nsjconf_t*, const char*)' at config.cc:
nsjail> /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/protob
nsjail>   272 |     return *cast<TypeHandler>(rep_->elements[index]);
nsjail>       |             ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
nsjail> /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/protob
nsjail> /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/protob
nsjail>   642 |     void* elements[(std::numeric_limits<int>::max() - 2 * sizeof(int)) /
nsjail>       |           ^~~~~~~~
nsjail> In member function 'const typename TypeHandler::Type& google::protobuf::internal
nsjail>     inlined from 'const Element& google::protobuf::RepeatedPtrField<T>::Get(int)
nsjail>     inlined from 'const nsjail::IdMap& nsjail::NsJailConfig::_internal_uidmap(in
nsjail>     inlined from 'const nsjail::IdMap& nsjail::NsJailConfig::uidmap(int) const'
nsjail>     inlined from 'bool config::parseInternal(nsjconf_t*, const nsjail::NsJailCon
nsjail>     inlined from 'bool config::parseFile(nsjconf_t*, const char*)' at config.cc:
nsjail> /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/protob
nsjail>   272 |     return *cast<TypeHandler>(rep_->elements[index]);
nsjail>       |             ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
nsjail> /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/protob
nsjail> /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/protob
nsjail>   642 |     void* elements[(std::numeric_limits<int>::max() - 2 * sizeof(int)) /
nsjail>       |           ^~~~~~~~
nsjail> cc1plus: all warnings being treated as errors
nsjail> make: *** [<builtin>: config.o] Error 1
nsjail> make: *** Waiting for unfinished jobs....
error: builder for '/nix/store/ax1vxmplpmahyf05ldcy1mwjzbc9yjk3-nsjail-3.4.drv' failed w
ith exit code 2;
       last 25 log lines:
       >     inlined from 'bool config::parseInternal(nsjconf_t*, const nsjail::NsJailCo
       >     inlined from 'bool config::parseFile(nsjconf_t*, const char*)' at config.cc
       > /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/proto
       >   272 |     return *cast<TypeHandler>(rep_->elements[index]);
       >       |             ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
       > /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/proto
       > /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/proto
       >   642 |     void* elements[(std::numeric_limits<int>::max() - 2 * sizeof(int))
       >       |           ^~~~~~~~
       > In member function 'const typename TypeHandler::Type& google::protobuf::interna
       >     inlined from 'const Element& google::protobuf::RepeatedPtrField<T>::Get(int
       >     inlined from 'const nsjail::IdMap& nsjail::NsJailConfig::_internal_uidmap(i
       >     inlined from 'const nsjail::IdMap& nsjail::NsJailConfig::uidmap(int) const'
       >     inlined from 'bool config::parseInternal(nsjconf_t*, const nsjail::NsJailConfig&)' at config.cc:198:68,
       >     inlined from 'bool config::parseFile(nsjconf_t*, const char*)' at config.cc:331:20:
       > /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/protobuf/repeated_ptr_field.h:272:30: error: array subscript -1 is below array bounds of 'void* [268435454]' [-Werror=array-bounds=]
       >   272 |     return *cast<TypeHandler>(rep_->elements[index]);
       >       |             ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
       > /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/protobuf/repeated_ptr_field.h: In function 'bool config::parseFile(nsjconf_t*, const char*)':
       > /nix/store/f5nrc42s6g7rqhwvx8d22v4qkz7a4ymj-protobuf-21.12/include/google/protobuf/repeated_ptr_field.h:642:11: note: while referencing 'google::protobuf::internal::RepeatedPtrFieldBase::Rep::elements'
       >   642 |     void* elements[(std::numeric_limits<int>::max() - 2 * sizeof(int)) /
       >       |           ^~~~~~~~
       > cc1plus: all warnings being treated as errors
       > make: *** [<builtin>: config.o] Error 1
       > make: *** Waiting for unfinished jobs....
       For full logs, run 'nix log /nix/store/ax1vxmplpmahyf05ldcy1mwjzbc9yjk3-nsjail-3.4.drv'.
```

### Additional context

Add any other context about the problem here.

### Notify maintainers

@arcz @bosu  @c0bw3b 

let me know if there is anything I can do to help.

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
- system: `"x86_64-linux"`
 - host os: `Linux 6.6.8, NixOS, 24.05 (Uakari), 24.05.20240116.f032654`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.19.2`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/tools/security/nsjail/default.nix
+++ b/pkgs/tools/security/nsjail/default.nix
@@ -18,6 +18,8 @@ stdenv.mkDerivation rec {
   buildInputs = [ libnl protobuf protobufc ];
   enableParallelBuilding = true;
 
+  env.NIX_CFLAGS_COMPILE = toString [ "-Wno-error" ];
+
   preBuild = ''
     makeFlagsArray+=(USER_DEFINES='-DNEWUIDMAP_PATH=${shadow}/bin/newuidmap -DNEWGIDMAP_PATH=${shadow}/bin/newgidmap')
   '';

